cpp/mcap/include/mcap/mcap.hpp

```diff
@@ -224,57 +327,201 @@ struct MetadataIndex {
+   * access is desired.
+   */
+  bool noSummary;
+  /**
+   * @brief Target uncompressed Chunk payload size in bytes. Once a Chunk's
+   * uncompressed data meets or exceeds this size, the Chunk will be compressed
+   * (if compression is enabled) and written to disk. Note that smaller Chunks
+   * may be written, such as the last Chunk in the Data section. This option is
+   * ignored if `noChunking=true`.
+   */
   uint64_t chunkSize;
+  /**
+   * @brief Compression algorithm to use when writing Chunks. This option is
+   * ignored if `noChunking=true`.
+   */
   Compression compression;
+  /**
+   * @brief Compression level to use when writing Chunks. Slower generally
+   * produces smaller files, at the expense of more CPU time. These levels map
+   * to different internal settings for each compression algorithm.
+   */
   CompressionLevel compressionLevel;
+  /**
+   * @brief The recording profile. See
+   * <https://github.com/foxglove/mcap/tree/main/docs/specification/profiles>
+   * for more information on well-known profiles.
+   */
   std::string profile;
+  /**
+   * @brief A freeform string written by recording libraries. For this library,
+   * the default is "libmcap {Major}.{Minor}.{Patch}".
+   */
   std::string library;
-  KeyValueMap metadata;
 
   McapWriterOptions(const std::string_view profile)
       : noChunking(false)
-      , noIndexing(false)
+      , noSummary(false)
       , chunkSize(DefaultChunkSize)
       , compression(Compression::None)
       , compressionLevel(CompressionLevel::Default)
       , profile(profile)
       , library("libmcap " LIBRARY_VERSION) {}
 };
 
+/**
+ * @brief An abstract interface for reading MCAP data.
+ */
 struct IReadable {
   virtual inline ~IReadable() = default;
 
+  /**
+   * @brief Returns the size of the file in bytes.
+   *
+   * @return uint64_t The total number of bytes in the MCAP file.
+   */
   virtual uint64_t size() const = 0;
+  /**
+   * @brief This method is called by MCAP reader classes when they need to read
+   * a portion of the file.
+   *
+   * @param output A pointer to a pointer to the buffer to write to. This method
+   *   is expected to either maintain an internal buffer, read data into it, and
+   *   update this pointer to point at the internal buffer, or update this
+   *   pointer to point directly at the source data if possible. The pointer and
+   *   data must remain valid and unmodified until the next call to read().
+   * @param offset The offset in bytes from the beginning of the file to read.
+   * @param size The number of bytes to read.
+   * @return uint64_t Number of bytes actually read. This may be less than the
+   *   requested size if the end of the file is reached. The output pointer must
+   *   be readable from `output` to `output + size`. If the read fails, this
+   *   method should return 0.
+   */
   virtual uint64_t read(std::byte** output, uint64_t offset, uint64_t size) = 0;
```

Comment:
`std::byte const** output` ?

Reply:
```
mcap.inl:1430:27: error: cannot initialize a parameter of type 'const std::byte **' with an rvalue of type 'std::byte **'
  bytesRead = reader.read(&record->data, offset + 9, record->dataSize);
                          ^~~~~~~~~~~~~
```

Reply:
Why is record->data a pointer to mutable bytes?

Reply:
I think there are a few places with const incorrectness that need to be cleaned up.